### ISSUE TITLE
Inconsistency: Meta `description` character length

### ISSUE THREAD

**Issue by stefanvd** [hober/web-extensions#218]
When creating a new browser extension you need to have a product name and description text. However, there is inconsistency with the length of the description text in the `manifest.json` file. As a result, there is no universal browser extension description text that is common for all web browser platforms.

**Google Chrome** (= Microsoft Edge = Firefox)
In Google Chrome the `description` text is limited to **132** characters:

> A plain text string (no HTML or other formatting; no more than 132 characters) 

Source: https://developer.chrome.com/docs/extensions/mv2/manifest/description/

**Safari**
In Safari the `description` text is limited to **112** characters:
<img width="1154" alt="Screenshot 2022-05-28 at 14 33 59" src="https://user-images.githubusercontent.com/20127018/170826063-c9d8a21f-f7a1-430d-92c3-b56fb324fa4f.png">

See the screenshot here above.

**Resources:**
- https://developer.chrome.com/docs/extensions/mv2/manifest/description/
- https://wiki.mozilla.org/Marketplace/Add-ons/Validation

**Comment by carlosjeurissen** [comment]
Nice catch! It seems the limit of 132 is not enforced in the Chrome browser. I was able to load an extension with a `description` length of 2500. This applies to the `name` and `short_name` properties as well. Funny note, is the minimum length. For the `name` and `description` properties, an empty string with length 0 is allowed. Yet, `short_name` requires at least one character.

Unlike the browser, the Chrome Web Store actually does enforce certain limits. Which could be the 132 `description` length. For the `name` and `short_name` this is a maximum of 45 characters.

@stefanvd the error from Safari, is it when loading the extension or when trying to submit it to their store? Conclusion: It's the app store which rejects such extensions.

### Going forward
For extension developers and stores, it's very much preferred to have clear documented limits and preferably synchronized across browsers and stores.

Please note the values could be localised using the __MSG_ID__ syntax. Translated strings in some languages can be a lot longer in terms of length compared to English.

We should clearly figure out and document what each restriction and requirement at this moment of time is in each extension store and browser. **This can then be followed with a safe length range which extension developers should be able to rely on**.

Additionally, we should consider emojis being part of the value for these properties. As well as special characters like ™. These are not supported at the moment in the Whale extension store. See: https://forum.whale.naver.com/topic/39748/.

It would be very helpful to get more documentation on the difference between requirements from the browser and those from extension stores.

### Manifest v4 / extension store proposal
To make sure current manifest v2 and v3 extensions can be loaded fine in new environments, adding additional restrictions only makes sense in manifest v4 or as extension store upload requirement.

My suggestion is to require at least a `description` and `short_name` length of 1, otherwise the properties can just be skipped. This is if the `description` property is really optional (see unresolved issue https://github.com/w3c/webextensions/issues/100).

The `short_name` length should not exceed the `name` length when both don't contain a localised variable. Since this would indicate a developer error.

It would also make a lot of sense when the `name` property requires at least a length of 1. Otherwise we should make the extension name an optional property (which can be considered).

Lastly, we can explore increasing the character length and synchronize it between browsers and stores.

### Clear example how long values are not considered
<img width="705" alt="lorem-ipsum" src="https://user-images.githubusercontent.com/1038267/171133063-56c74c09-4063-41dd-a7fb-c851c0ff4234.png">

**Comment by stefanvd** [comment]
@carlosjeurissen Yes, that is on submitting to the AppStoreConnect. When the processing is done at the Apple server, I receive an email that said, "has one or more issues".
